projects/components/src/datagrid/interfaces/datagrid-column.interface.ts

```diff
@@ -40,6 +40,106 @@ export enum GridColumnSortDirection {
  */
 export type FunctionRenderer<T> = (record: T) => string;
 
+/**
+ * A type of button who's displayability does not depend on the selected entity.
+ */
+export interface GlobalButton {
+    /**
+     * The translated text of the button.
+     */
+    label: string;
+    /**
+     * The function that is called when the button is pressed.
+     */
+    handler: () => void;
+    /**
+     * The function that is called to determine if the button should be displayed.
+     */
+    shouldDisplay: () => boolean;
+    /**
+     * The css class the button should have.
+     */
+    class?: string;
+}
+
+/**
+ * A type of button who's displayability dependends on the selected entity.
+ */
+export interface ContextualButton<R> {
```

Comment:
GlobalButton and ContextualButton seems to have same properties except for id and Icon. How about making a common interface like...
GridButton {
 ...common properties
 shouldDisplay: (recs?: R[]) => boolean;
 handler(entity?: R[]) => void;
}

ContextualButton can then extend the interface

Reply:
I extracted the label and class to the common interface. I chose not to extract the handler or shouldDisplay function to the interface because that would require giving GlobalButton the ability to accept records in the callback functions and a generic type which felt a little awkward. What do you think about this?